cmd/lncli/cmd_open_channel.go

```diff
@@ -460,6 +460,22 @@ func openChannel(ctx *cli.Context) error {
 			"combination with the --psbt flag")
 	}
 
+	// In case we dont want to wait until the channel is fully confirmed
+	// onchain we use the `OpenChannelSync` rpc call.
```

Comment:
tried to explain it in the other comment:

> 
Otherwise the for loop on the LND server is still running although the client already exited after receiving the PendingOpenRequest. So it is a slight improvement because the server side will only cancel the goroutine when the Channel is open (has the required confirmations).

verified it on regtest, we indeed keep the goroutine of the RPC server open even if we opened a channel in an "unblock" way, thats why we should use the sync call in this case:

```
goroutine 896 [select, 6 minutes]:
github.com/lightningnetwork/lnd.(*rpcServer).OpenChannel(0x1400018ca00, 0x14000d17e60, {0x1024e69d0, 0x1400074a140})
        github.com/lightningnetwork/lnd/rpcserver.go:2408 +0x2a4
github.com/lightningnetwork/lnd/lnrpc._Lightning_OpenChannel_Handler({0x1024a6120, 0x1400018ca00}, {0x1024e1f70, 0x140023c78c0})
        github.com/lightningnetwork/lnd/lnrpc/lightning_grpc.pb.go:2392 +0x120
github.com/grpc-ecosystem/go-grpc-prometheus.init.(*ServerMetrics).StreamServerInterceptor.func4({0x1024a6120, 0x1400018ca00}, {0x1024e1748, 0x14002862000}, 0x140025933f8, 0x1024b2fc0)
        github.com/grpc-ecosystem/go-grpc-prometheus@v1.2.0/server_metrics.go:121 +0xc8
```

Reply:
moved to https://github.com/lightningnetwork/lnd/pull/8934